src/tribler/core/utilities/search_utils.py

```diff
@@ -27,3 +32,194 @@ def split_into_keywords(string, to_filter_stopwords=False):
 
 def filter_keywords(keywords):
     return [kw for kw in keywords if len(kw) > 0 and kw not in DIALOG_STOPWORDS]
+
+
+def item_rank(query: str, item: dict):
+    title = item['name']
+    seeders = item.get('num_seeders', 0)
+    leechers = item.get('num_leechers', 0)
+    freshness = time.time() - item.get('updated', 0)
+    return torrent_rank(query, title, seeders + leechers * 0.1, freshness)
+
+
+def torrent_rank(query: str, title: str, seeders: int = 0, freshness: Optional[float] = 0) -> float:
+    """
+    Calculates search rank for a torrent.
+    Takes into account:
+      - similarity of the title to the query string;
+      - the reported number of seeders;
+      - how long ago the torrent file was created.
+    """
+    freshness = max(0, freshness or 0)
+    tr = title_rank(query or '', title or '')
+    sr = (seeders_rank(seeders or 0) + 9) / 10  # range [0.9, 1]
+    fr = (freshness_rank(freshness) + 9) / 10  # range [0.9, 1]
+    result = tr * sr * fr
+    # uncomment the next line to debug the function inside an SQL query:
+    # print(f'*** {result} : {seeders}/{freshness} ({freshness / SECONDS_IN_DAY} days)/{title} | {query}')
+    return result
+
+
+def seeders_rank(seeders: float) -> float:
+    """
+    Calculates rank based on the number of seeders. The result is normalized to the range [0, 1]
+    """
+    return seeders / (100 + seeders)  # inf seeders -> 1; 100 seeders -> 0.5; 10 seeders -> approx 0.1
```

Comment:
The plot above with the seeders range [-400, 400] is misleading. The number of seeders is a positive value, usually in the range [0, 1000].

I want the `seeders_rank` to return a value in the range [0, 1]. It should return 0 for a torrent with zero seeders and 1 for a torrent with an infinite number of seeders.

So, the actual graph for a function looks like this. With zero seeders, the torrent has `seeders_rank` 0; with 100 seeders it goes up to 0.5, and then slowly increases to 1.0 when the number of seeders reaches infinity. It does not have singularities like on your previous graph.

<img width="1148" alt="Screenshot 2022-10-24 at 17 33 08" src="https://user-images.githubusercontent.com/2931136/197566230-9bcdf225-aff7-466c-b2ad-80af6b4d42d5.png">

It is possible to use different curves with the similar shape: `x / (N + x)`, `2 * arctan(x / N) / PI`, `1 - exp(x * ln(0.5) / N)`, but it is not actually matter on practice

<img width="1189" alt="Screenshot 2022-10-24 at 18 34 56" src="https://user-images.githubusercontent.com/2931136/197578739-aebb2a83-b4af-4623-afbc-244962e3243b.png">

I use the simplest and most efficient function with desired properties: zero rating for zero seeders, 0.5 rating for 100 seeders, 1 rating for an infinite number of seeders, and soft curve.

Reply:
Thank you for the explanation. 
It would be nice if they were placed in the code. It follows from your explanation that `100` is not a randomly picked number, but a number with meaning. 
That's why it also may be reflected in the code:

```python
def seeders_rank(seeders: float, half_point: float = 100) -> float:
    """ Calculates rank based on the number of seeders.

    This function has desired properties: zero rating for zero seeders, 0.5 rating for `half_point` seeders,
    1 rating for an infinite number of seeders, and a soft curve.
    
    The result is normalized to the range [0, 1].

    Args:
        seeders:  The number of seeders. It is a positive value, usually in the range [0, 1000]
        half_point: The number of seeders at which the rank should return half of the maximum value, i.e. 0.5

    Returns: float in the range [0, 1]
    """

    # It is possible to use different curves with the similar shape: 
    # 2 * arctan(x / N) / PI, 
    # 1 - exp(x * ln(0.5) / N)
    # but it does not actually matter in practice
    return seeders / (half_point + seeders)
```

Reply:
I added the description to the formula and extracted a `SEEDERS_HALF_RANK` named constant

Reply:
🙏